specify/specify7
# Correctly enforce Collector -> agent uniqueness rule

Fixes #3841 

TO TEST: 
- Try adding a Collector to a CollectingEvent who is already a Collector in that event
- Test and make sure other 'types' of uniqueness rules work correctly (examples provided below)
  - Catalognumber must be unique in Collection 
  - Role and Agent must be unique to AccessionAgent

This PR also sets the uniqueIdentifier field in the CollectionObject, CollectingEvent, and Locality tables to be globally unique.
